src/Service/DigitalPostHelper.php

```diff
@@ -72,6 +77,106 @@ public function sendDigitalPost(string $cpr, string $name, Address $address, str
     }
 
+    public function createDigitalPost(Document $document, string $subject, string $entityType, Uuid $entityId, array $digitalPostAttachments, array $digitalPostRecipients): void
+    {
+        $digitalPosts = [];
+
+        // Handle attachments and add extra digital post entities if necessary due to size or number of attachments
+        // Size limit for attachments and main document combined is 80mb
+        // The max number of attachments per digital post is 10
+        $documentPath = $this->getFilePath($document);
+        $currentSizeOfDigitalPost = $mainDocumentSize = filesize($documentPath);
+        $currentNumberOfAttachments = 0;
+        $sizeLimit = 80 * (10 ** 6);
+        $attachmentLimit = 10;
+
+        // Setup first DigitalPost
+        $currentDigitalPost = new DigitalPostEntity();
+
+        // Handle non-attachment properties on DigitalPost
+        foreach ($digitalPostRecipients as $recipient) {
+            assert($recipient instanceof DigitalPostEntity\Recipient);
+            $currentDigitalPost->addRecipient($recipient);
+        }
+
+        $currentDigitalPost->setDocument($document);
+        $currentDigitalPost->setEntityType($entityType);
+        $currentDigitalPost->setEntityId($entityId);
+
+        // Subject might depend on attachments in case multiple digital posts are needed,
+        // but we add it here and modify it later if necessary
+        $currentDigitalPost->setSubject($subject);
+
+        $this->entityManager->persist($currentDigitalPost);
+
+        if (count($digitalPostAttachments) > 0) {
+            foreach ($digitalPostAttachments as $attachment) {
+                assert($attachment instanceof DigitalPostAttachment);
+                $attachmentSize = filesize($this->getFilePath($attachment->getDocument()));
+                // Attachments are made from documents i.e. we are guaranteed they are less than 80mb
+
+                // Check if adding attachment would violate restrictions
+                if ($currentSizeOfDigitalPost + $attachmentSize >= $sizeLimit || $currentNumberOfAttachments >= $attachmentLimit) {
+                    // Persist current digital post entity and make new for current attachment
+                    $digitalPosts[] = $currentDigitalPost;
+
+                    $currentDigitalPost = new DigitalPostEntity();
+
+                    // Handle non-attachment properties on DigitalPost
+                    foreach ($digitalPostRecipients as $recipient) {
+                        $currentDigitalPost->addRecipient($recipient);
+                    }
+
+                    $currentDigitalPost->setDocument($document);
+                    $currentDigitalPost->setEntityType($entityType);
+                    $currentDigitalPost->setEntityId($entityId);
+                    $currentDigitalPost->setSubject($subject);
+
+                    $this->entityManager->persist($currentDigitalPost);
+                    $this->entityManager->flush();
```

Comment:
Any specific reason for flushing more than once. You could just flush outside the loop.